### DISCUSSION TITLE
getEloquentQuery not working as expected

### DISCUSSION THREAD

**Discussion by flashmediasolutions** [filamentphp/filament#6592] [Help]
Having a problem using getEloquentQuery. I have placed it in a resource's main file and I am attempting to display only the resource records owned by the logged in user. 

After throwing in a dd() I can verify It is not being fired when displaying the main resource record list, but it is being fired when viewing a single record. If the user owns the record, I can view it. If not, it will show a 403.

Am I missing something? Does getEloquentQuery need to be called earlier in the process to impact the main record list?

```
public static function getEloquentQuery(): Builder
    {
        return parent::getEloquentQuery()->whereHas('users', function ($query) {
            $query->where('users.id', auth()->id());
        });
    }
```

Using latest Filament / Laravel.

**Comment by flashmediasolutions** [comment]
Placed this in the resources List page. Not sure if this is the correct way. I assumed the getEloquentQuery would impact all resource queries.

```
public function getTableQuery(): Builder
    {
        return parent::getTableQuery()->whereHas('users', function ($query) {
            $query->where('users.id', auth()->id());
        });
    }
```

**Comment by flashmediasolutions** [comment]
New wrinkle.

Now I can see the records for a user, but I can't display all records for user with 'view_any' permissions.

```
public function getTableQuery(): Builder
    {
        $user = auth()->user();
        if (!$user->can('view_any_post')) {
            return parent::getTableQuery();
        }

        return parent::getTableQuery()->whereHas('users', function ($query) {
            $query->where('users.id', auth()->id());
        });
    }
```

**Comment by danharrin** [comment]
Hmm, `getEloquentQuery()` should be the base of every query made by the resource, on all pages. Something must be wrong somewhere, none of this sounds in line with how I think it works. If it's a bug, please open an issue with a *minimal* reproduction repository.

**Comment by flashmediasolutions** [comment]
@danharrin tested a fresh install and worked as expected. I will comb through my code to find out where it breaks that functionality. Might be tied to use of Vemto. 

Thanks.

**Comment by flashmediasolutions** [comment]
For anyone who finds this experiencing the same issue (and using Vemto), the fix is to remove:

```
use HasDescendingOrder;
```

from your resource's "List" page. It is added automatically by Vemto and conflict with the getEloquentQuery function.
